core-bundle/src/Resources/contao/classes/BackendUser.php

```diff
@@ -582,6 +583,58 @@ public function getRoles()
 
 		return $this->roles;
 	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function serialize()
+	{
+		$data = array
+		(
+			'id' => $this->id,
+			'username' => $this->username,
+			'password' => $this->password,
+			'admin' => $this->admin,
+			'disable' => $this->disable,
+			'start' => $this->start,
+			'stop' => $this->stop
+		);
+
+		return serialize($data);
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function unserialize($serialized)
+	{
+		$data = unserialize($serialized, array('allowed_classes'=>false));
+
+		if (array_keys($data) != array('id', 'username', 'password', 'admin', 'disable', 'start', 'stop'))
+		{
+			return;
+		}
+
+		list($this->id, $this->username, $this->password, $this->admin, $this->disable, $this->start, $this->stop) = array_values($data);
+	}
```

Comment:
instead of duplicating the whole code, we should do what Symfony usually does:

```php
public function serialize()
{
    return serialize(['admin' => $this->admin, 'parent' => parent::serialize()];
}

public function unserialize($serialized)
{
    $data = unserialize($serialized, array('allowed_classes'=>false));

    if (array_keys($data) != array('admin', 'parent'))
    {
        return;
    }

    list($this->admin, $parent) = array_values($data);

    parent::unserialize($parent);
}
```

Reply:
Changed in 3b5e2382037a28a4170e094c915b2645419034a3.